apps/worker/src/app/shared/utils/should-halt-on-step-failure.ts

```diff
@@ -0,0 +1,27 @@
+import { StepTypeEnum } from '@novu/shared';
+
+export const shouldHaltOnStepFailure = (
+  stepType: StepTypeEnum | undefined,
+  shouldStopOnFail: boolean | undefined
+): boolean | undefined => {
+  if (!stepType) {
```

Comment:
Is there a scenario where a job doesn't have a stepType?

Reply:
should not be, but it is optional and i do not want to take a risk..